application/src/objects/obstacles/abstract-dynamic-obstacle.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+var Phaser = require('phaser');
+
+var AbstractDynamicObstacle = function(state, x, y, key, collisionGroup)
+{
+    Phaser.Sprite.apply(this, [state.game, x, y, key]);
+
+    this.createPhysicsBody(state);
+
+    this.body.setCollisionGroup(collisionGroup);
+    this.body.mass = this.getAttributes().mass;
+    this.body.angularDamping = this.getAttributes().angularDamping;
+};
+
+AbstractDynamicObstacle.prototype = Object.create(Phaser.Sprite.prototype);
+
+AbstractDynamicObstacle.prototype.getSpritePath = function()
+{
+    throw new Error('Attempted to load assets on abstract class');
+};
+
+AbstractDynamicObstacle.prototype.getAttributes = function()
```

Comment:
I think ice and powerups would apply modifiers. The base constant would still apply.

Yeah, I would use all caps in the key names.